### PR TITLE
Feature/api intergration

### PR DESCRIPTION
**Pull request by MosDevx**
## Hey There!! :wave: 

Kindly review my PR

### Changes Implemented:
- intergrated a Books Api
- load books from Api
- add & delete books from Api

### Thank you :smiley:

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,36 +1,100 @@
...
+  const data = await response.json();
+  // console.log(data);
+  // for (const [key, value] of Object.entries(data)) {
+
+  Object.entries(data).forEach(([key, value]) => {
+    booksArray.push({
+      item_id: key, title: value[0].title, author: value[0].author, category: value[0].category,
+    });
+  });
+  // }
```
Comment (TadesseDev) [inline src/redux/books/booksSlice.js]: - [x] Dear @MosDevx, though your code is clean, we still need to remove commented codes from our codebase. this makes your code messy.

File: src/Components/Book.jsx [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 
 	const onRemoveButtonClicked = (id)=>{
-		// console.log('RemoveButtonCalled',id)
-		dispatch(deleteBook(id))
+		dispatch(deleteBookApi(id))
 	}
+
+	// const onEditButtonClicked = ()=>{
+	// 	dispatch(fetchBooks())
+	// }
```
Comment (TadesseDev) [inline src/Components/Book.jsx]: - [x] Dear @MosDevx, As mentioned in my other comment for a sec of clean code please remove commented codes.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @MosDevx 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
EquatorialVictoriousBlackpanther-size_restricted

## Highlights
 - [ ] Good job overall following the template design ✅
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/Book.jsx (hunk 2 above)
Comment (MosDevx) [reply src/Components/Book.jsx]: ### Thanks @Tadesse-Alemayehu  :smile:  and Merry Christmas :partying_face:

Comment (TadesseDev) [reply src/Components/Book.jsx]: You're welcome @MosDevx, Merry Christmas 🙌

**Review by TadesseDev — APPROVED** [review]
### Hi @MosDevx 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
68747470733a2f2f6669636f666f72756d732e6d796669636f2e636f6d2f74352f696d6167652f736572766572706167652f696d6167652d69642f343932363069463839303835363833444139334135422f696d6167652d73697

## Highlights
 - [ ] Good job making the changes requested previously. ✅
 - [ ]  Correct Gitflow Flow. ✅
 - [ ]  Good commit history. ✅
 - [ ]  Good PR title/description. ✅
 - [ ]  All linters passed ✅
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**